xla/README.md

```diff
@@ -0,0 +1,124 @@
+zip -r local_changes.zip . -x  ".git/*" "checkpoints/*" "data/*" "out/*"
+# Copy the .zip file to the TPU VM
+gcloud compute tpus tpu-vm scp --worker=all local_changes.zip "lit-gpt:~"
+# Unzip on each host
+gcloud compute tpus tpu-vm ssh lit-gpt --worker=all --command="cd ~; unzip -q -o local_changes.zip"
+
+# Example of a typical workflow
+gcloud compute tpus tpu-vm ssh tmp --worker=all --command="cd ~; bash install_dependencies.sh"
+gcloud compute tpus tpu-vm ssh tmp --worker=all --command="cd ~; bash prepare_checkpoints.sh"
+gcloud compute tpus tpu-vm ssh tmp --worker=all --command="cd ~; bash run_desired_script.sh"
+
+# This will allow you to kill all python processes on all workers
+gcloud compute tpus tpu-vm ssh tmp --worker=all --command="pkill -e python"
+```
+
+For the rest of this tutorial, it will be assumed that it is being run on a single host for simplicity.
+
+</details>
+
+Once inside the machine, clone the repository and install the dependencies:
+
+```shell
+git clone https://github.com/Lightning-AI/lit-gpt
+cd lit-gpt
+pip install -r requirements.txt
+```
+
+Install Optimized BLAS:
+
+```shell
+sudo apt update
+sudo apt install libopenblas-dev
+```
+
+Since Lit-GPT requires a torch version newer than torch 2.0.0, manually install nightly builds of torch and torch_xla:
+
+```shell
+pip install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch-nightly-cp38-cp38-linux_x86_64.whl
+pip install https://storage.googleapis.com/tpu-pytorch/wheels/tpuvm/torch_xla-nightly-cp38-cp38-linux_x86_64.whl
+```
+
+While computations will run by default using the new PjRT runtime, it is recommended to set the following environment variables:
+
+```shell
+export ALLOW_MULTIPLE_LIBTPU_LOAD=1
+export PJRT_DEVICE=TPU
+```
+
+> [!NOTE]\
+> An extensive guide on setup and available options can be found [here](https://cloud.google.com/tpu/docs/v4-users-guide).
+
+Since a new machine was created, you may need to download pretrained weights.
+They can be copied to the machine using `gcloud compute tpus tpu-vm scp`, or you can follow the steps described in our [downloading guide](download_stablelm.md).
+
+It is also recommended to set up a persistent disk from which to load checkpoints.
+Follow [this guide](https://cloud.google.com/tpu/docs/setup-persistent-disk#setting_up_a_tpu_vm_and_a_persistent_disk) to do so.
+Read-write disks are not supported in multihost VM setups, so persistent disks cannot be used to save checkpoints in that case.
+Persistent disks can still be useful in read-only mode to load pretrained weights before finetuning or inference.
+In multihost settings, FSDP will save checkpoint shards per host and consolidate them into a single checkpoint.
+For safekeeping, it is recommended to upload the consolidated checkpoints to a Google Cloud bucket.
+Alternatively, you can use the `scp` command to transfer these checkpoints from the TPU VM periodically, although this is not implemented in our scripts.
+
+
+## Inference
```

Comment:
Is there a reason we are not including pretraining?

Reply:
We'd have to get numbers for it